proposals/0047-syscall-and-sysvar-for-last-restart-slot.md

```diff
@@ -0,0 +1,148 @@
+---
+simd: '0047'
+title: Syscall and Sysvar for last restart slot
+authors:
+  - Godmode Galactus (Mango Markets)
+category: Standard
+type: Core
+status: Draft
+created: 2023-04-15
+---
+
+## Summary
+
+Create a new syscall and sysvar that can be used to get the last restart slot.
+
+A new Sysvar :
+`SysvarLastRestartS1ot1111111111111111111111`
+
+Method signature for syscall:
+`fn sol_get_last_restart_slot() -> Slot`
+
+## Motivation
+
+In Solana, when the cluster cannot reach consensus, it is currently restarted
+using `ledger-tool` with a hard fork on a slot. All participating nodes need to
+restart their validator specifying the hard fork slot. Once all nodes
+participating in the restart effort exceed 80% of the total stakes, the restart
```

Comment:
isnt stake a dynamic unit, what if some validators plan to drop staking and the new stake is less than 70% of the older stake

Reply:
There are also some CLI arguments which help to disable stakes on nodes which do not participate in restart effort.

Reply:
@ripatel-jump @godmodegalactus thanks gotcha, was just curious